middlewares/retry.go

```diff
@@ -56,9 +63,12 @@ func (retry *Retry) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func isNetworkError(status int) bool {
-	return status == http.StatusBadGateway || status == http.StatusGatewayTimeout
-}
+// NetErrorCtxKey is a custom type that is used as key for the request context.
+// Using an own type we avoid potential collisions with other keys that may have the same string name.
+type NetErrorCtxKey string
```

Comment:
Thanks for the hint, I refactored it now in this way. Now the logic for mutating the context is solely in the retry middleware and therefore we can use a private type 🎉